core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -601,13 +606,13 @@ public static FieldStats readFrom(StreamInput in) throws IOException {
     public static String typeName(byte type) {
         switch (type) {
             case 0:
-                return "whole-number";
+                return "integer";
```

Comment:
Could you move this method up with its brother methods, above all the subclasses?